Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -1022,11 +1022,21 @@ void RenderGrid::placeAutoMajorAxisItemOnGrid(RenderBox& gridItem, AutoPlacement
 
 GridTrackSizingDirection RenderGrid::autoPlacementMajorAxisDirection() const
 {
+    if (isMasonryColumns())
+        return ForColumns;
+    if (isMasonryRows())
+        return ForRows;
+
     return style().isGridAutoFlowDirectionColumn() ? ForColumns : ForRows;
 }
 
 GridTrackSizingDirection RenderGrid::autoPlacementMinorAxisDirection() const
 {
+    if (isMasonryColumns())
```

Comment:
Could replace this whole function with `autoPlacementMajorAxisDirection() == ForColumns ? ForRows : ForColumns` to avoid duplicating logic.

Reply:
Sounds good